src/schemes/fluid/surface_tension.jl

```diff
@@ -93,49 +53,53 @@ struct SurfaceTensionAkinci{ELTYPE} <: AkinciTypeSurfaceTension
+#     return cohesion_force_akinci(surface_tension, smoothing_length, m_b, pos_diff, distance)
+# end
+
+# function (surface_tension::SurfaceTensionAkinci)(support_radius, m_b, na, nb, pos_diff,
+#                                                  distance)
+#     (; surface_tension_coefficient) = surface_tension
+#     return cohesion_force_akinci(surface_tension, support_radius, m_b, pos_diff,
+#                                  distance) .- (surface_tension_coefficient * (na - nb))
+# end
+
+# Note that `floating_point_number^integer_literal` is lowered to `Base.literal_pow`.
+# Currently, specializations reducing this to simple multiplications exist only up
+# to a power of three, see
+# https://github.com/JuliaLang/julia/blob/34934736fa4dcb30697ac1b23d11d5ad394d6a4d/base/intfuncs.jl#L327-L339
+# By using the `@fastpow` macro, we are consciously trading off some precision in the result
+# for enhanced computational speed. This is especially useful in scenarios where performance
+# is a higher priority than exact precision.
+@fastpow @inline function cohesion_force_akinci(surface_tension, support_radius, m_b,
+                                                pos_diff, distance)
     (; surface_tension_coefficient) = surface_tension
 
     # Eq. 2
-    # We only reach this function when distance > eps
+    # We only reach this function when distance > eps and `distance < support_radius`
     C = 0
-    if distance <= support_radius
-        if distance > 0.5 * support_radius
-            # Attractive force
-            C = (support_radius - distance)^3 * distance^3
-        else
-            # distance < 0.5 * support_radius
-            # Repulsive force
-            C = 2 * (support_radius - distance)^3 * distance^3 - support_radius^6 / 64.0
-        end
-        C *= 32.0 / (pi * support_radius^9)
+    if distance > 0.5 * support_radius
+        # Attractive force
+        C = (support_radius - distance)^3 * distance^3
+    else
+        # `distance < 0.5 * support_radius`
+        # Repulsive force
+        C = 2 * (support_radius - distance)^3 * distance^3 - support_radius^6 / 64.0
     end
+    C *= 32.0 / (pi * support_radius^9)
 
     # Eq. 1 in acceleration form
-    cohesion_force = -surface_tension_coefficient * mb * C * pos_diff / distance
+    cohesion_force = -surface_tension_coefficient * m_b * C * pos_diff / distance
 
     return cohesion_force
 end
 
-@inline function adhesion_force_akinci(surface_tension::AkinciTypeSurfaceTension,
-                                       support_radius, mb, pos_diff, distance,
-                                       adhesion_coefficient)
+@inline function adhesion_force_akinci(surface_tension, support_radius, m_b, pos_diff,
+                                       distance, adhesion_coefficient)
     # Eq. 7
-    # We only reach this function when distance > eps
+    # We only reach this function when `distance > eps`
```

Comment:
```suggestion
    # We only reach this function when `sqrt(eps()) < distance < support_radius`
```
And the if condition below is not needed anymore.

Reply:
That is not correct since the value of this function is not defined for distance=support_radius. The comments are inaccurate its actually `<=`.

Reply:
yes but the part in the if becomes negative and so the result of the ^0.25 is a complex number.

Reply:
Why negative? It only becomes zero, no?

Reply:
Yes the result should be 0 instead of the complex number

Reply:
The part inside the root becomes zero, and `0.0^0.25` is `0.0`. Or am I missing something?

Reply:
Yes it should be but it doesn't. The simulations crashes because the value is negative if we remove the special handling.